### ISSUE TITLE
sanity.openjdk jdk_net_0 httpserver/Test9a crash libcrypto

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#8373]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/1
jdk_net_0
TEST: com/sun/net/httpserver/Test9a.java
```
12:26:03  rerun:
12:26:03  cd /home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/TKG/test_output_15797134779504/jdk_net_0/work/scratch/0 && \
12:26:03  HOME=/home/jenkins \
12:26:03  LANG=en_CA.UTF-8 \
12:26:03  PATH=/bin:/usr/bin:/usr/sbin \
12:26:03      /home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdkbinary/j2sdk-image/bin/javac \
12:26:03          -J-ea \
12:26:03          -J-esa \
12:26:03          -J-Xmx512m \
12:26:03          -J-Xcompressedrefs \
12:26:03          -J-Dtest.class.path.prefix=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/TKG/test_output_15797134779504/jdk_net_0/work/classes/1/com/sun/net/httpserver:/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/openjdk/openjdk-jdk/jdk/test/com/sun/net/httpserver \
12:26:03          -J-Dtest.file=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/openjdk/openjdk-jdk/jdk/test/com/sun/net/httpserver/Test9a.java \
12:26:03          -J-Dtest.src=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/openjdk/openjdk-jdk/jdk/test/com/sun/net/httpserver \
12:26:03          -J-Dtest.src.path=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/openjdk/openjdk-jdk/jdk/test/com/sun/net/httpserver \
12:26:03          -J-Dtest.classes=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/TKG/test_output_15797134779504/jdk_net_0/work/classes/1/com/sun/net/httpserver \
12:26:03          -J-Dtest.class.path=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/TKG/test_output_15797134779504/jdk_net_0/work/classes/1/com/sun/net/httpserver \
12:26:03          -J-Dtest.vm.opts='-ea -esa -Xmx512m -Xcompressedrefs' \
12:26:03          -J-Dtest.tool.vm.opts='-J-ea -J-esa -J-Xmx512m -J-Xcompressedrefs' \
12:26:03          -J-Dtest.compiler.opts= \
12:26:03          -J-Dtest.java.opts= \
12:26:03          -J-Dtest.jdk=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdkbinary/j2sdk-image \
12:26:03          -J-Dcompile.jdk=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdkbinary/j2sdk-image \
12:26:03          -J-Dtest.timeout.factor=8.0 \
12:26:03          -J-Dtest.root=/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/openjdk/openjdk-jdk/jdk/test \
12:26:03          -d /home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/TKG/test_output_15797134779504/jdk_net_0/work/classes/1/com/sun/net/httpserver \
12:26:03          -sourcepath /home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/openjdk/openjdk-jdk/jdk/test/com/sun/net/httpserver \
12:26:03          -classpath /home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/openjdk/openjdk-jdk/jdk/test/com/sun/net/httpserver:/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/TKG/test_output_15797134779504/jdk_net_0/work/classes/1/com/sun/net/httpserver:/home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdkbinary/j2sdk-image/lib/tools.jar \
12:26:03          -XDignore.symbol.file=true /home/jenkins/workspace/Test_openjdk8_j9_sanity.openjdk_ppc64le_linux_Personal/openjdk-tests/openjdk/openjdk-jdk/jdk/test/com/sun/net/httpserver/Test9a.java
12:26:03  
12:26:03  ACTION: main -- Failed. Unexpected exit from test [exit code: 255]
12:26:03  REASON: User specified action: run main/othervm Test9a 
12:26:03  TIME:   5.259 seconds
12:26:03  messages:
12:26:03  command: main Test9a
12:26:03  reason: User specified action: run main/othervm Test9a 
12:26:03  Mode: othervm [/othervm specified]
12:26:03  elapsed time (seconds): 5.259
12:26:03  configuration:
12:26:03  STDOUT:
12:26:03  Test9a: 
12:26:03  STDERR:
12:26:03  Unhandled exception
12:26:03  Type=Segmentation error vmState=0x00040000
12:26:03  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000002
12:26:03  Handler1=00003FFF78C074D0 Handler2=00003FFF78A9F6C0
12:26:03  R0=FFFFFFFFCA62C1D6 R1=00003FFF28BCAEE0 R2=00003FFF293F7800 R3=00003FFE6C003300
12:26:03  R4=00003FFF3C02FFFE R5=03FFFFFFFFB4F2DB R6=0000000000000010 R7=0000000AAD4515DD
12:26:03  R8=0000000812BD72FE R9=00000000998116E5 R10=00000000D1950063 R11=00000000A6E9D890
12:26:03  R12=00000007C88A1D8F R13=00003FFF28BD6900 R14=0000000000000412 R15=00000009D40C823B
12:26:03  R16=0000000000000000 R17=0000000000000000 R18=00000000998116E5 R19=00000000D1950063
12:26:03  R20=00003FFF28BCAF10 R21=0000000000000000 R22=0000000000000000 R23=0000000000000000
12:26:03  R24=0000000000000000 R25=0000000000000000 R26=0000000000000000 R27=0000000000000000
12:26:03  R28=0000000000000000 R29=0000000000000000 R30=0000000000000000 R31=0000000000000000
12:26:03  NIP=00003FFF29220448 MSR=800000010280D033 ORIG_GPR3=00000000000081C8 CTR=0000000000000010
12:26:03  LINK=00003FFF29220434 XER=0000000020000000 CCR=000000004424284C SOFTE=0000000000000001
12:26:03  TRAP=0000000000000300 DAR=00003FFF3C030000 dsisr=0000000040000000 RESULT=0000000000000000
12:26:03  FPR0 6ab4e4653424d0d3 (f: 874827968.000000, d: 1.048047e+206)
12:26:03  FPR1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR2 3ff0000000000000 (f: 0.000000, d: 1.000000e+00)
12:26:03  FPR3 3c50000000000000 (f: 0.000000, d: 3.469447e-18)
12:26:03  FPR4 40262e42fefa3800 (f: 4277811200.000000, d: 1.109035e+01)
12:26:03  FPR5 0000001000000008 (f: 8.000000, d: 3.395193e-313)
12:26:03  FPR6 0000000000000200 (f: 512.000000, d: 2.529616e-321)
12:26:03  FPR7 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR8 c219aeb13ede4320 (f: 1054753536.000000, d: -2.757626e+10)
12:26:03  FPR9 8aebfe6ae20912a9 (f: 3792245504.000000, d: -4.660965e-256)
12:26:03  FPR10 d43fca472b22815f (f: 723681600.000000, d: -6.790334e+97)
12:26:03  FPR11 4040000000000000 (f: 0.000000, d: 3.200000e+01)
12:26:03  FPR12 0000000000000002 (f: 2.000000, d: 9.881313e-324)
12:26:03  FPR13 4025971f230cef38 (f: 588050240.000000, d: 1.079516e+01)
12:26:03  FPR14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR16 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR17 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR19 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
12:26:03  FPR30 4040000000000000 (f: 0.000000, d: 3.200000e+01)
12:26:03  FPR31 4040000000000000 (f: 0.000000, d: 3.200000e+01)
12:26:03  Module=/lib/powerpc64le-linux-gnu/libcrypto.so.1.0.0
12:26:03  Module_base_address=00003FFF291B0000
12:26:03  Target=2_90_20200122_655 (Linux 4.4.0-141-generic)
12:26:03  CPU=ppc64le (16 logical CPUs) (0x1fe0b0000 RAM)
12:26:03  ----------- Stack Backtrace -----------
12:26:03  (0x00003FFF29220448 [libcrypto.so.1.0.0+0x70448])
12:26:03  (0x00003FFF293AD338 [libcrypto.so.1.0.0+0x1fd338])
12:26:03  SHA1_Update+0x198 (0x00003FFF2921E908 [libcrypto.so.1.0.0+0x6e908])
12:26:03  (0x00003FFF292D4F5C [libcrypto.so.1.0.0+0x124f5c])
12:26:03  EVP_DigestUpdate+0x24 (0x00003FFF292CAF14 [libcrypto.so.1.0.0+0x11af14])
12:26:03  (0x00003FFF292C3A84 [libcrypto.so.1.0.0+0x113a84])
12:26:03  (0x00003FFF292C3DE0 [libcrypto.so.1.0.0+0x113de0])
12:26:03  RAND_bytes+0x58 (0x00003FFF292C4738 [libcrypto.so.1.0.0+0x114738])
12:26:03  BN_rand+0xc4 (0x00003FFF29260674 [libcrypto.so.1.0.0+0xb0674])
12:26:03  (0x00003FFF29260170 [libcrypto.so.1.0.0+0xb0170])
12:26:03  BN_BLINDING_create_param+0xd4 (0x00003FFF292620F4 [libcrypto.so.1.0.0+0xb20f4])
12:26:03  RSA_setup_blinding+0x1f0 (0x00003FFF29296AD0 [libcrypto.so.1.0.0+0xe6ad0])
12:26:03  (0x00003FFF29289F3C [libcrypto.so.1.0.0+0xd9f3c])
12:26:03  (0x00003FFF2928B330 [libcrypto.so.1.0.0+0xdb330])
12:26:03  RSA_private_decrypt+0x28 (0x00003FFF292967B8 [libcrypto.so.1.0.0+0xe67b8])
12:26:03  Java_jdk_crypto_jniprovider_NativeCrypto_RSADP+0xdc (0x00003FFF40154D5C [libjncrypto.so+0x4d5c])
12:26:03  (0x00003FFF78CE3478 [libj9vm29.so+0x183478])
12:26:03  (0x00003FFF78CE1454 [libj9vm29.so+0x181454])
12:26:03  (0x00003FFF78B84644 [libj9vm29.so+0x24644])
12:26:03  (0x00003FFF78C3A4CC [libj9vm29.so+0xda4cc])
12:26:03  (0x00003FFF78BEB614 [libj9vm29.so+0x8b614])
12:26:03  (0x00003FFF78C5B370 [libj9vm29.so+0xfb370])
12:26:03  (0x00003FFF78AA0AC8 [libj9prt29.so+0x30ac8])
12:26:03  (0x00003FFF78C56A04 [libj9vm29.so+0xf6a04])
12:26:03  (0x00003FFF78B41308 [libj9thr29.so+0x11308])
12:26:03  (0x00003FFF79908070 [libpthread.so.0+0x8070])
12:26:03  clone+0x98 (0x00003FFF797E3A70 [libc.so.6+0x123a70])
12:26:03  ---------------------------------------
```

**Comment by pshipton** [comment]
@ashbm5 can you pls take a look

**Comment by ashbm5** [comment]
@shanchao95 please try to reproduce

**Comment by shanchao95** [comment]
I tried to re-run the tests with same parameters on Grinder, and tried to run the test9a.java locally with jvm from `https://140-211-168-230-openstack.osuosl.org/artifactory/ci-eclipse-openj9/Build_JDK8_ppc64le_linux_Personal/655/OpenJ9-JDK8-ppc64le_linux-20200122-162742.tar.gz`. Did not see it occur again. Here is tests results on Grinder `https://ci.adoptopenjdk.net/job/Grinder/2157/`.

**Comment by pshipton** [comment]
On the open machines, grinders on ub16-ppcle-2 (1x), ub16-ppcle-3 (5x) didn't reproduce it, but grinders on ub16p8j92 and ub16p8j95 did.
https://ci.eclipse.org/openj9/view/Test/job/Grinder/631/
https://ci.eclipse.org/openj9/view/Test/job/Grinder/632/

**Comment by pshipton** [comment]
@jdekonin are there openssl updates not installed on the ub16p8j9x machines? If you do update the machines, pls record the original level and the updated level.

**Comment by jdekonin** [comment]
Updates to both sets of machines were going to occur on a scheduled "maintenance 1/2 day".  But that date hadn't been set.
ub16-ppcle-1 -> ub16-ppcle-3 are on 1.0.2g-1ubuntu4.15, the systems have 135 packages to be updated, 6 are security updates.
ub16p8j91 -> ub16p8j96 are on 1.0.2g-1ubuntu4, the systems have 384 packages to be updated, 161 of which are security updates.

If you are in agreement that these can be rotated through tomorrow @rajdeep-singh and I can proceed, just advise if we should be doing an full system update to bring them all current, or if we should just be performing an openssl update.

**Comment by pshipton** [comment]
We can do a full system update, but let's only do one system of each ppcle and p8j9 first and make sure they are still working fine afterward before doing the rest.

**Comment by jdekonin** [comment]
ub16p8j91 and ub16-ppcle-2 have been updated to the latest levels and brought back online.  Note that ub16-ppcle-2 also had an issue with protobuf.

**Comment by pshipton** [comment]
Confirmed the crash still occurs on the upgraded ub16p8j91
https://ci.eclipse.org/openj9/view/Test/job/Grinder/638

**Comment by pshipton** [comment]
@jdekonin what is the openssl version after the update?

**Comment by jdekonin** [comment]
ub16p8j91 is at 1.0.2g-1ubuntu4.15, same as ub16-ppcle-2.

**Comment by M-Davies** [comment]
This is also occurring at Adopt. Last nights [JDK14/j9/aix](https://ci.adoptopenjdk.net/job/Test_openjdk14_j9_sanity.openjdk_ppc64_aix/19/) run
```
Stacktrace
Unexpected exit from test [exit code: 255]
    
Standard Output
Test9: 
    
Standard Error

Unhandled exception
Type=Segmentation error vmState=0x00040000
J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000032
Handler1=09001000A0D13978 Handler2=09001000A0CE9640
R0=000000005A827999 R1=000001002D3CFAB0 R2=09001000A0AA11D0 R3=00000100280FCC20
R4=09001000A0AAFFE4 R5=03FFFFFFFFFFFFFD R6=0000000066044000 R7=000000017567C67A
R8=000000015CCC1A9F R9=00000000E67A4B7F R10=0000000057048A60 R11=00000000FFE54C48
R12=00000000998353EB R13=000001002D3DB800 R14=00000000305618A0 R15=0000000077044860
R16=0000000000000000 R17=0000000000000000 R18=0000000000000000 R19=0000000000000000
R20=0000000000000000 R21=0000000000000000 R22=0000000000000000 R23=0000000000000000
R24=0000000000000000 R25=0000000000000000 R26=0000000000000000 R27=0000000000000000
R28=0000000000000000 R29=0000000000000000 R30=0000000000000000 R31=0000000000000000
IAR=0900000006F2DE5C LR=0900000006F2DC74 MSR=A00000000200D032 CTR=03FFFFFFFFFFFD62
CR=24402244000100F4 FPSCR=8200200000000000 XER=000100F482002000
FPR0 4040000000000000 (f: 0.000000, d: 3.200000e+01)
FPR1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR2 3fec4ed454000000 (f: 1409286144.000000, d: 8.846227e-01)
FPR3 433000000005c0c0 (f: 377024.000000, d: 4.503600e+15)
FPR4 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR5 433000000000a9ec (f: 43500.000000, d: 4.503600e+15)
FPR6 4117030000000000 (f: 0.000000, d: 3.770240e+05)
FPR7 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR8 bea0145980b6318c (f: 2159424000.000000, d: -4.792062e-07)
FPR9 bf56aefc8fb3c300 (f: 2410922752.000000, d: -1.384493e-03)
FPR10 402c62a66ab53432 (f: 1790260224.000000, d: 1.419268e+01)
FPR11 3fd55555555450ef (f: 1431589120.000000, d: 3.333333e-01)
FPR12 4530000000000000 (f: 0.000000, d: 1.934281e+25)
FPR13 4042800000000000 (f: 0.000000, d: 3.700000e+01)
FPR14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR16 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR17 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR19 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
FPR31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
Module=/usr/lib/libcrypto.a
Module_base_address=0900000006F07180
Target=2_90_20200312_33 (AIX 7.1)
CPU=ppc64 (16 logical CPUs) (0x200000000 RAM)
----------- Stack Backtrace -----------
(0x0900000006F2DB80 [libcrypto.a+0x26a00])
(0x0900000006F4563C [libcrypto.a+0x3e4bc])
(0x0900000006F3C24C [libcrypto.a+0x350cc])
(0x0900000006F44380 [libcrypto.a+0x3d200])
(0x0900000006F43AF4 [libcrypto.a+0x3c974])
(0x0900000006F43290 [libcrypto.a+0x3c110])
(0x0900000006F47FEC [libcrypto.a+0x40e6c])
(0x0900000006F47D50 [libcrypto.a+0x40bd0])
(0x0900000006F472A8 [libcrypto.a+0x40128])
(0x0900000006F46354 [libcrypto.a+0x3f1d4])
(0x0900000006F8C1CC [libcrypto.a+0x8504c])
(0x0900000006F8D2FC [libcrypto.a+0x8617c])
(0x0900000006F465D8 [libcrypto.a+0x3f458])
Java_jdk_crypto_jniprovider_NativeCrypto_RSADP+0xe4 (0x090000000FD8ACC8 [libjncrypto.so+0x5cc8])
(0x0900000007BB66BC [libj9vm29.so+0x14f6bc])
ffi_call+0x98 (0x0900000007BB59BC [libj9vm29.so+0x14e9bc])
(0x0900000007BC0C24 [libj9vm29.so+0x159c24])
(0x0900000007ADEF7C [libj9vm29.so+0x77f7c])
runJavaThread+0x1d8 (0x0900000007AC9FBC [libj9vm29.so+0x62fbc])
javaProtectedThreadProc+0x11c (0x0900000007A699E0 [libj9vm29.so+0x29e0])
omrsig_protect+0x4a4 (0x0900000007D61DC8 [libj9prt29.so+0x58dc8])
javaThreadProc+0x68 (0x0900000007A6984C [libj9vm29.so+0x284c])
thread_wrapper+0x558 (0x0900000007CEA63C [libj9thr29.so+0x563c])
_pthread_body+0xf0 (0x0900000000570E14 [libpthreads.a+0x3e14])
---------------------------------------
```

**Comment by pshipton** [comment]
Occurrences in jdk11 runs:
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.openjdk_ppc64le_linux_Nightly/3
com/sun/net/httpserver/Test9.java.Test9
com/sun/net/httpserver/Test9a.java.Test9a

**Comment by M-Davies** [comment]
Occurred in a JDK14-j9-aix run last night https://ci.adoptopenjdk.net/view/Test_openjdk/job/Test_openjdk14_j9_sanity.openjdk_ppc64_aix/31/

Output files: https://ibm.box.com/s/kkvbuy03z3za24ocugw8owhppynkym7w

**Comment by pshipton** [comment]
There is a fix underway for this https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/381
It needs to be ported to the other jdk versions.

**Comment by pshipton** [comment]
This is an intermittent problem tracked in a Milestone. I'm going to dissolve the "Blocking sanity.openjdk" project since the less intermittent problems were resolved by reconfiguring sanity.openjdk to tier1.

**Comment by pshipton** [comment]
The fix https://github.com/eclipse/openj9/issues/8373#issuecomment-603814836 is merged, but not yet into the 0.20.0 branch.

**Comment by pshipton** [comment]
The updates to the 0.20.0 branches are complete.
